core/src/com/po/fuck/weapons/Bullet.java

```diff
@@ -38,11 +41,13 @@ public void update(float delta) {
         List<Collidable> collidableList = All.collidableCollection.collides(polygon);
         for (Collidable collidable : collidableList) {
             if (collidable instanceof Entity) {
-                ((Entity) collidable).takeDamage(1);
-                Manager.destroy_raw(this);
+                if (this.tryDamage((Entity) collidable)) {
+                    ((Entity) collidable).takeDamage(1);
```

Comment:
I think it should be `tryDamage(enemy)` and only then destroying bullet, also after cast rename `collidable` into `enemy`.